### PR TITLE
Feature/OECD Data Pipeline

### PR DESCRIPTION
**Pull request by ThatcherK**
ticket #647 
The csvs generated from this have been pushed to gates/oecd branch on the di-website-data repo

### CODE REVIEW

**Comment by ThatcherK** [comment]
@dean-breed Please take a look at this data update script for the gates OECD csv data

**Review by dean-breed — DISMISSED** [review]
This all looks good. Thanks @ThatcherK !

**Comment by dean-breed** [comment]
@ThatcherK I will now be working on creating the final dataset 'donor-by-recip-2019.csv' which I will add to this script to push to gh too

**Review by edwinmp — CHANGES_REQUESTED** [review]
Why do you have the CSV files in here? If those are the files generated by running this script, then I'd rather they don't get committed - add the entire folder to `.gitignore`

Won't merge this in until @dean-breed's pending work on it is done.

**Comment by dean-breed** [comment]
I have added the new CSV code @edwinmp

**Comment by dean-breed** [comment]
But I need to change one thing... Wait up @edwinmp

**Comment by dean-breed** [comment]
Done @edwinmp

File: data_updates/Python/oecd_data.py
Comment (edwinmp) [inline data_updates/Python/oecd_data.py]: Test the generated files on the `di-chart-boilerplate` repo (`page/gates` branch) to make sure the tables and charts still function, then change this to `main`.

**Review by edwinmp — CHANGES_REQUESTED** [review]
@ThatcherK still seeing the CSV files. Once they're committed, you have to delete them and commit before that `.gitignore` change takes effect.

**Comment by edwinmp** [comment]
Lastly, this isn't plugged into any pre-existing update process so it won't deploy updates automatically ... there's an `oecd.sh` but I'm not sure if that's part of a regular update pipeline ... @wakibi do we have a scheduled event or such script for auto updating OECD data?

**Comment by ThatcherK** [comment]
@dean-breed , i was comparing the recipient by donor table form the previous csv, with the current and noticed a slight difference, for united states, reproductive health care and family planning. Previously the first donor was `Unspecified` , its now `Bilateral, unspecified`. Is this the desired result?

**Comment by dean-breed** [comment]
@ThatcherK Thanks for pointing this out. I think it's fine but Mattie and Duncan are taking a look today and I'll make sure they agree!

**Comment by dean-breed** [comment]
@wakibi Good to go, I think!

**Comment by wakibi** [comment]
> Lastly, this isn't plugged into any pre-existing update process so it won't deploy updates automatically ... there's an `oecd.sh` but I'm not sure if that's part of a regular update pipeline ... @wakibi do we have a scheduled event or such script for auto updating OECD data?

@edwinmp not one I know of. @dean-breed, is this the intended functionality? A one-off or manual update not tied to any ETL process?

**Comment by dean-breed** [comment]
@wakibi We can just trigger this script when the DAC data is updated

**Comment by edwinmp** [comment]
> @wakibi We can just trigger this script when the DAC data is updated

@dean-breed @ThatcherK  then you're going to need to create a `.sh` script that we can configure to trigger manually on via the scheduled events feature ... similar to [this](https://github.com/devinit/ddw-analyst-ui/blob/develop/data_updates/iati_transactions.sh)

**Comment by dean-breed** [comment]
@ThatcherK are you okay to do this?

**Comment by ThatcherK** [comment]
@dean-breed  sure I'll look into it

File: data_updates/Python/oecd_data.py
Comment (wakibi) [inline data_updates/Python/oecd_data.py]: @ThatcherK sorry this is coming late, but would be nice to have the folder 'oecd_csv' created on the fly if it does not exist already. Otherwise everything will crush (or at the very least handle it gracefully). Also, can we have the file names as one word? Use underscores ("_") in case you really want to keep it descriptive. Sometimes filenames with spaces can behave unexpectedly.

File: data_updates/Python/oecd_data.py
Comment (wakibi) [inline data_updates/Python/oecd_data.py]: Same as above

File: data_updates/Python/oecd_data.py
Comment (wakibi) [inline data_updates/Python/oecd_data.py]: Same as above

File: data_updates/Python/oecd_data.py
Comment (dean-breed) [reply data_updates/Python/oecd_data.py]: I'll leave @ThatcherK to do the folder creation but wanted to say that the only reason they are the file names is because that's what DK called them originally. I agree that underscores are better

File: data_updates/Python/oecd_data.py
Comment (wakibi) [inline data_updates/Python/oecd_data.py]: For the constants here that may need changing... Can we find a better way to handle them? You have hard-coded the URLs for example and they are staging URLs, are they the same ones that will be used on production? The Remote folder shows 2022, shall we maintain 2022 next year as well or we shall update it to 2023? If the year will keep changing, I propose using the datetime functions to get the current year. For the URLs, possibly we use Environment variables and fall back to the staging URLs if no environment variable has been found.

**Review by wakibi — CHANGES_REQUESTED** [review]

**Comment by ThatcherK** [comment]
@wakibi @edwinmp , all changes made. I now have to change csv names on the gates repo, to match the ones with underscores...

**Comment by ThatcherK** [comment]
@wakibi  and @edwinmp , please can we find an agreed format for the naming of the csv files, originally they were space separated words, Chris suggested underscore separated. What would be preferable?

**Comment by edwinmp** [comment]
@edwinmp go with whatever Chris has recommended ... just be sure to update everything that will be affected by that change (several places) ... get in a huddle and I'll talk you through it.

**Comment by ThatcherK** [comment]
@wakibi , should this be merged?

**Review by wakibi — DISMISSED** [review]
LGTM

**Review by edwinmp — DISMISSED** [review]

File: data_updates/Python/oecd_data.py
Comment (edwinmp) [inline data_updates/Python/oecd_data.py]: I think the default can be added directly in the `getenv` function i.e
`os.getenv('PURPOSE_CODE_TRENDS_URL', 'https://staging-ddw.devinit.org/api/export/1241/')`

... do for the others as wells

**Review by edwinmp — DISMISSED** [review]

**Comment by ThatcherK** [comment]
Tested on staging, everything looks good now!! I need an approving review to merge

**Review by edwinmp — APPROVED** [review]

**Pull request merged**
